tests/test_agents.py

```diff
@@ -817,3 +824,27 @@ async def test_interactive(dummy_env: DummyEnv, mocker):
     next_obs, _, done, _ = await dummy_env.step(action.value)  # noqa: RUF059
 
     assert mock_input.call_count >= 2
+
+
+@pytest.mark.asyncio
+async def test_sliding_window():
+    """Test that sliding window limits message history."""
+    # Create a simple agent with sliding_window=1
+    agent = SimpleAgent(sliding_window=1)
+    env = DummyEnv()
+
+    # Initialize
+    obs, tools = await env.reset()
+    state = await agent.init_state(tools)
+
+    for _ in range(5):
+        action, state, _ = await agent.get_asv(state, obs)
+        obs, _, _, _ = await env.step(action.value)
+
+    # Check that old messages are hidden
+    messages = state.get_next_state(obs).messages
+    assert len(messages) == 4
+    assert messages[0].role == "user"
+    assert messages[1].content == "[Previous messages - hidden]"
+    assert isinstance(messages[2], ToolRequestMessage)
+    assert isinstance(messages[3], ToolResponseMessage)
```

Comment:
Should we assert on the content in all of the messages? These assertions seem somewhat unspecific, maybe add failure messages explaining their rationale

```python
assert messages[1].content == "[Previous messages - hidden]", "Expected a hidden message because XYZ"
```